lib/checkout_sdk/api_client.rb

```diff
@@ -127,11 +127,21 @@ def parse_response(response)
       raise CheckoutApiException, response if response.status < 200 || response.status >= 300
 
       metadata = CheckoutUtils.map_to_http_metadata(response)
-      body = (JSON.parse(response.body, object_class: OpenStruct) if !response.body.nil? && response.body != '')
+      body = parse_json_or_contents(response)
       body = OpenStruct.new if body.nil?
       body = OpenStruct.new(items: body) if body.is_a? Array
       body.metadata = metadata if body.is_a? OpenStruct
       body
     end
+
+    def parse_json_or_contents(response)
+      return unless !response.body.nil? && response.body != ''
```

Comment:
I feel that this would read better like:
`return if response.body.nil? && response.body == ''`

Reply:
And now I'm in doubts if the && condition is ok there

Reply:
That makes sense, there is no need to unless here, and also it should be an `or` as well, good catch